src/text/writer.rs

```diff
@@ -259,7 +291,21 @@ impl<W: Write> TextWriter<W> {
 
     /// Writes the provided byte array slice as an Ion clob.
     pub fn write_clob(&mut self, _value: &[u8]) -> IonResult<()> {
-        todo!()
+        // clob_value to be written based on defined STRING_ESCAPE_CODES.
+        let mut clob_value = String::from("");
+        for i in 0.._value.len() {
+            let c = (_value[i] & 0xff) as char;
+            let escaped_byte = &self.string_escape_codes[c as usize];
+            if escaped_byte != "" {
+                clob_value = format!("{}{}", clob_value, escaped_byte);
+            } else {
+                clob_value = format!("{}{}", clob_value, &c);
```

Comment:
This is clever! Unfortunately, it requires `clob_value` to be reallocated in each loop iteration. Here are some ways you can append more text to the end of the existing string without making a new one:
* [`String::push`](https://doc.rust-lang.org/std/string/struct.String.html#method.push) will add the contents a given `char` to the end of an existing `String`.
* [`String::push_str`](https://doc.rust-lang.org/std/string/struct.String.html#method.push_str) will add the contents of a `&str` to the end of an existing `String`.
* [`write!`](https://doc.rust-lang.org/std/macro.write.html) also appends to the end of a given `String` (or other output), but allows you to use the same formatting flags as `format!` and `println!`.